File: android-gif-drawable/src/test/java/pl/droidsonroids/gif/GifOptionsTest.java
package pl.droidsonroids.gif;

import org.junit.Before;
import org.junit.Test;

import static org.assertj.core.api.Java6Assertions.assertThat;

public class GifOptionsTest {

	private GifOptions gifOptions;

	@Before
	public void setUp() {
		gifOptions = new GifOptions();
	}

	@Test
	public void testInitialValues() {
		assertThat(gifOptions.inSampleSize).isEqualTo((char) 1);
		assertThat(gifOptions.inIsOpaque).isFalse();
	}

	@Test
	public void setInSampleSize() throws Exception {
		gifOptions.setInSampleSize(2);
		assertThat(gifOptions.inSampleSize).isEqualTo((char) 2);
	}

	@Test
	public void setInIsOpaque() throws Exception {
		gifOptions.setInIsOpaque(true);
		assertThat(gifOptions.inIsOpaque).isTrue();
	}

	@Test
	public void copyFromNonNull() throws Exception {
		GifOptions source = new GifOptions();
		source.setInIsOpaque(false);
		source.setInSampleSize(8);
		gifOptions.setFrom(source);
		assertThat(gifOptions).isEqualToComparingFieldByField(source);
	}

	@Test
	public void copyFromNull() throws Exception {
		GifOptions defaultOptions = new GifOptions();
		gifOptions.setInIsOpaque(false);
		gifOptions.setInSampleSize(8);
		gifOptions.setFrom(null);
		assertThat(gifOptions).isEqualToComparingFieldByField(defaultOptions);
	}
}

File: android-gif-drawable/src/test/java/pl/droidsonroids/gif/GifOptionsTest.kt
package pl.droidsonroids.gif

import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test

class GifOptionsTest {

    private lateinit var gifOptions: GifOptions

    @Before
    fun setUp() {
        gifOptions = GifOptions()
    }

    @Test
    fun testInitialValues() {
        assertThat(gifOptions.inSampleSize).isEqualTo(1.toChar())
        assertThat(gifOptions.inIsOpaque).isFalse
    }

    @Test
    fun setInSampleSize() {
        gifOptions.setInSampleSize(2)

        assertThat(gifOptions.inSampleSize).isEqualTo(2.toChar())
    }

    @Test
    fun setInIsOpaque() {
        gifOptions.inIsOpaque = true

        assertThat(gifOptions.inIsOpaque).isTrue
    }

    @Test
    fun copyFromNonNull() {
        val source = GifOptions()

        source.inIsOpaque = false
        source.setInSampleSize(8)
        gifOptions.setFrom(source)

        assertThat(gifOptions).isEqualToComparingFieldByField(source)
    }

    @Test
    fun copyFromNull() {
        val defaultOptions = GifOptions()

        gifOptions.inIsOpaque = false
        gifOptions.setInSampleSize(8)
        gifOptions.setFrom(null)

        assertThat(gifOptions).isEqualToComparingFieldByField(defaultOptions)
    }
}